### PR TITLE
fix(chat): surface env-provider trust errors as system message before agent spawn

### LINKED ISSUE #478: bug: chat prompts silently fail in new workspaces when repo requires setup (e.g. mise trust) and no setup script is configured
## Summary
When creating a new workspace in a repo that requires environment setup (e.g. `mise trust`, `direnv allow`, `nix develop` priming), the agent inherits a broken/incomplete environment and chat prompts silently fail. There is no UI feedback explaining *why* — the user just sees prompts go nowhere until they remember to add a setup script in repo settings.

## Reproduction
1. Add a repo whose `.mise.toml` (or equivalent) is not yet trusted by the local mise install.
2. Do **not** configure a setup script under repo settings.
3. Create a new workspace.
4. Send a chat prompt → it hangs or fails with no actionable error.
5. Workaround: open repo settings, add `mise trust && mise install` (or equivalent) to the personal setup script field, enable auto-run, recreate workspace.

## Expected behavior
At least one of:
- The workspace-creation flow detects that the env-provider (mise/direnv/nix) reports an untrusted/unprimed state and surfaces a banner/modal *before* the user sends a prompt, with a one-click "configure setup script" affordance.
- Chat surfaces the env-provider error inline as a System message instead of failing silently.
- The mise/direnv plugins default `auto_trust` to a prompt-on-first-use rather than silent opt-in.

## Relevant code paths
- Setup runs are gated behind `skip_setup=true` by default at `src-tauri/src/commands/workspace.rs:129` and a per-repo `setup_script_auto_run` flag at `src-tauri/src/commands/workspace.rs:158-173`.
- Env-provider warmup already runs on repo add and is the natural detection point: `src-tauri/src/commands/repository.rs:114-121` (comment explicitly notes direnv/trust issues surface here).
- Mise plugin already has the detection logic but only acts when `auto_trust` is set: `plugins/env-mise/init.lua:28-36`.
- Setup-script UI form: `src/ui/src/components/settings/sections/RepoSettings.tsx:376-420` (placeholder already says `mise trust && mise install`, suggesting this is a known-common case).

## Environment
- Claudette branch: `eben/polished-orchid`
- OS: Linux (Wayland/X11)

### PR DESCRIPTION
**Pull request by bakedbean**
## Summary
- Adds `ResolvedEnv::trust_errors()` and `format_trust_message()` helpers that filter trust-related env-provider failures (`mise.toml not trusted`, `.envrc is blocked`, …) and render a markdown system message with a per-provider remediation hint.
- On chat send, posts that message into the chat session and emits a `chat-system-message` Tauri event so the open ChatPanel renders it immediately instead of waiting for the failing turn to finalize and trigger a history reload.
- Dedupe via a `posted_env_trust_warning` flag on `AgentSessionState`; the flag clears on the existing env-drift teardown so a new failure (e.g. user fixed mise but `.envrc` is now blocked) reposts exactly once.

Closes #478. Without this, a workspace whose mise/direnv config hasn't been allowed silently spawned the agent with a degraded env and the user saw no explanation for the resulting hang. The error string was already captured into `ResolvedSource.error` but never surfaced.

Generic export failures (malformed TOML, missing CLI binary) are deliberately excluded — telling the user to "run `mise trust`" when `mise` isn't installed would be misleading.

## Test plan
- [x] `cargo test --all-features` (744 passed)
- [x] `cargo clippy -p claudette -p claudette-server --all-targets` with `-Dwarnings`
- [x] `cargo fmt --all --check`
- [x] `cd src/ui && bunx tsc -b` — frontend typechecks
- [x] `cd src/ui && bun run test` — 1003 passed
- [x] New unit tests in `src/env_provider/mod.rs` cover: mise "not trusted", direnv "blocked", generic export failure (must NOT match), no-error case, and the formatted message body.
- [ ] **Manual UAT**: in a worktree with an untrusted `mise.toml` and no setup script, send a chat prompt and confirm the system warning appears immediately (before any agent output) with the `mise trust` remediation hint. Then run `mise trust` in the worktree, send another prompt, and confirm the warning does not repost.

### CODE REVIEW

File: src/env_provider/mod.rs [hunk 1]
```diff
@@ -123,6 +123,79 @@ impl ResolvedEnv {
...
+        }
+        let mut body = String::from(
+            "**Environment setup needed.** One or more env-provider plugins reported a trust/priming error. Until this is resolved, the agent will spawn without the tools and variables this provider would normally contribute, which can cause prompts to fail.\n",
+        );
+        for src in errors {
+            let display = display_name_for(&src.plugin_name);
+            let hint = remediation_hint(&src.plugin_name);
+            let err = src.error.as_deref().unwrap_or("");
+            body.push_str(&format!("\n- **{display}**: {err}\n  - {hint}\n"));
+        }
```
[the remark this answers is not in the document]
Comment (bakedbean) [reply src/env_provider/mod.rs:165]: Done in 6744d19. Plugin stderr now renders inside an indented fenced code block under each list item, with an 800-byte excerpt cap (UTF-8 char-boundary safe) so a runaway diagnostic dump can't blow out the chat surface. Added `excerpt_error` tests covering single-line, multi-line indent, truncation, and multibyte boundary cases.

File: src/env_provider/mod.rs [hunk 2]
```diff
@@ -123,6 +123,79 @@ impl ResolvedEnv {
...
+/// direnv emit when their config hasn't been allowed yet, plus the
+/// "permission denied" wording nix-devshell can return for an
+/// untrusted flake. Generic `export:` / `detect:` errors that don't
+/// contain any of these markers fall through and are not surfaced as
+/// trust warnings.
+fn is_trust_error(error: &str) -> bool {
+    let lower = error.to_ascii_lowercase();
+    ["not trusted", "is blocked", "is not allowed", "untrusted"]
+        .iter()
+        .any(|needle| lower.contains(needle))
```
[the remark this answers is not in the document]
Comment (bakedbean) [reply src/env_provider/mod.rs:210]: Aligned the doc comment to the actual heuristic in 6744d19, rather than expanding the matcher. "permission denied" is intentionally NOT included — it's too generic and would catch unrelated POSIX filesystem failures (e.g. the plugin couldn't read its own config because of file perms), which would tell the user to run `mise trust` for an unrelated problem. The new comment calls that out explicitly.

File: src-tauri/src/state.rs [hunk 3]
```diff
@@ -115,6 +115,12 @@ pub struct AgentSessionState {
     /// turn. Updated each time a new user message is inserted; cleared on
     /// session teardown. See `agent_mcp_sink::ChatBridgeSink`.
     pub last_user_msg_id: Option<String>,
+    /// Set after we've posted a trust-error system message into the
+    /// chat for the current resolved env. Cleared whenever the resolved
+    /// env vars change (the user ran `direnv allow` / `mise trust`,
+    /// edited the config, toggled a provider) so a fresh failure
+    /// re-emits exactly once.
```
[the remark this answers is not in the document]
Comment (bakedbean) [reply src-tauri/src/state.rs]: Done in 6744d19. Used your suggested wording — the field doc now correctly describes the reset as scoped to the persistent-session drift teardown, not every env-vars change. Sticky-until-restart is the intended behavior so we don't spam the chat while the user is mid-fix.

**Comment by bakedbean** [comment]
image
